### ISSUE TITLE
FileUploader prop "files"

### ISSUE THREAD

**Issue by Tihomir971** [carbon-design-system/carbon-components-svelte#1112]
1. When I check property "files", it is type Array in my app. In documentation it is File[]. Btw, should it be FileList like in vanilla JS?

2. When on:change function is called, "files" property is still not assigned and i have to use event.target.files as prop for my function (Papa Parse).

I am a beginner, so suggestions (examples) would be welcome.

**Comment by metonym** [comment]
It is intentional that the native `FileList` from `e.target.files` is converted into an array.

However, you're correct in that the `on:change` is not consistent with `on:add` or `on:remove`.

I think it's reasonable that `on:change` be made consistent across the board.

**Comment by metonym** [comment]
Fixed in [v0.58.4](https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.58.4)

**Comment by metonym** [comment]
@Tihomir971 You can access `files` by binding to the prop or through the `on:change` event:

https://svelte.dev/repl/b3d0c2753b1e4c3989397e73e7d220b2?version=3.46.4

```svelte
<script>
  import { FileUploader } from "carbon-components-svelte";

  let files = [];

  $: console.log("files", files);
</script>

<FileUploader
  multiple
  labelTitle="Upload files"
  buttonLabel="Add files"
  labelDescription="Add files."
  status="complete"
  bind:files
  on:change={(e) => {
    console.log("change", e.detail);
  }}
/>

```
